src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs

```diff
@@ -1404,22 +1404,25 @@ private static Vector512<float> FusedMultiplyAdd(Vector512<float> x, Vector512<f
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static unsafe Vector128<float> LoadRemainderMaskSingleVector128(int validItems) =>
-            Vector128.LoadUnsafe(
-                ref Unsafe.As<uint, float>(ref MemoryMarshal.GetReference(RemainderUInt32Mask_16x16)),
-                (uint)((validItems * 16) + 12)); // last four floats in the row
+            Vector128.ConditionalSelect(
+                Vector128.LessThan(Vector128.Create(3, 2, 1, 0), Vector128.Create(validItems)).AsSingle(),
+                Vector128<float>.AllBitsSet,
+                Vector128<float>.Zero);
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static unsafe Vector256<float> LoadRemainderMaskSingleVector256(int validItems) =>
-            Vector256.LoadUnsafe(
-                ref Unsafe.As<uint, float>(ref MemoryMarshal.GetReference(RemainderUInt32Mask_16x16)),
-                (uint)((validItems * 16) + 8)); // last eight floats in the row
+            Vector256.ConditionalSelect(
+                Vector256.LessThan(Vector256.Create(7, 6, 5, 4, 3, 2, 1, 0), Vector256.Create(validItems)).AsSingle(),
+                Vector256<float>.AllBitsSet,
+                Vector256<float>.Zero);
 
 #if NET8_0_OR_GREATER
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static unsafe Vector512<float> LoadRemainderMaskSingleVector512(int validItems) =>
-            Vector512.LoadUnsafe(
-                ref Unsafe.As<uint, float>(ref MemoryMarshal.GetReference(RemainderUInt32Mask_16x16)),
-                (uint)(validItems * 16)); // all sixteen floats in the row
+            Vector512.ConditionalSelect(
```

Comment:
So for AVX512 it'd looked like 
```cs
var mmask16 mask = (1 << n) - 1;
var maskedAnd = Avx512.And(vec1, vec2, mask);
```
but we decided not to expose mask registers so if we expose these as a public API, we can intrinsify at leas the AVX512 version to be cheap 🙂